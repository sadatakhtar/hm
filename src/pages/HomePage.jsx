import React from "react";
import Button from "@mui/material/Button";
import { useNavigate } from "react-router-dom";

function HomePage() {
  const navigate = useNavigate();
  const handleBtn = () => {
    navigate("/about");
  };
  return (
    <div>
      <h1>Home Page</h1>
      <Button variant="contained" onClick={handleBtn}>
        Go to About
      </Button>
    </div>
  );
}

export default HomePage;
