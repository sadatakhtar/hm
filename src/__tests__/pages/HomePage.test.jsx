import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { BrowserRouter as Router } from "react-router-dom";
import { useSelector } from "react-redux";
import HomePage from "../../pages/HomePage";

// Mock useSelector
jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useSelector: jest.fn(),
}));

describe("HomePage component", () => {
  beforeEach(() => {
    useSelector.mockReturnValue("test value");
  });

  test("renders the HomePage component correctly", () => {
    const { asFragment } = render(
      <Router>
        <HomePage />
      </Router>
    );
    expect(screen.getByText("Home Page")).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  test("renders the button correctly", () => {
    const { asFragment } = render(
      <Router>
        <HomePage />
      </Router>
    );
    expect(screen.getByRole("button", { name: /Go to About/i })).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});