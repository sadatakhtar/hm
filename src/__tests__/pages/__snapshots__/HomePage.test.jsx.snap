// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage component renders the HomePage component correctly 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Home Page
    </h1>
    <p>
      Welcome to the home page
    </p>
    <p>
      Redux toolkit is implemented, Redux state value: test value
    </p>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-1abvrni-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Go to About
    </button>
  </div>
</DocumentFragment>
`;

exports[`HomePage component renders the button correctly 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Home Page
    </h1>
    <p>
      Welcome to the home page
    </p>
    <p>
      Redux toolkit is implemented, Redux state value: test value
    </p>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-1abvrni-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Go to About
    </button>
  </div>
</DocumentFragment>
`;
